plugins/dragdropselected.lua

```diff
@@ -1,177 +1,263 @@
+  return true
+end -- DocView:dnd_isInSelection
+
 
-  -- hand off to 'old' on_mouse_moved()
-  on_mouse_moved(self, x, y, ...)
-  -- override cursor as needed
-  if sCursor then self.cursor = sCursor end
+local on_mouse_moved = DocView.on_mouse_moved
+function DocView:on_mouse_moved(x, y, ...)
+  if not config.plugins.dragdropselected.enabled or not self.dnd_sText then
+    -- there is nothing to do -> hand off to original on_mouse_moved()
+    return on_mouse_moved(self, x, y, ...)
+  end
 
+  -- not sure we need to do this or if we better not
+  DocView.super.on_mouse_moved(self, x, y, ...)
+  if self.dnd_bDragging then
+    -- remove last caret showing insert location
+    self.doc:remove_selection(self.doc.last_selection)
+  else
+    self.dnd_bDragging = true
+    -- show that we are dragging something
+    self.cursor = 'hand'
+    -- make sure selection is marked
+    self.doc:set_selection(table.unpack(self.dnd_lSelection))
+  end
+  -- calculate line and column for current mouse position
+  local iLine, iCol = self:resolve_screen_position(x, y)
+  -- show insert location (unfortunately it doesn't always show, even when
+  -- calling draw_caret)
+  self.doc:add_selection(iLine, iCol)
+  -- update scroll position, if needed
+  self:scroll_to_line(iLine, true)
 end -- DocView:on_mouse_moved
 
--- override DocView:on_mouse_pressed
+
 local on_mouse_pressed = DocView.on_mouse_pressed
 function DocView:on_mouse_pressed(button, x, y, clicks)
   local caught = DocView.super.on_mouse_pressed(self, button, x, y, clicks)
   if caught then
       return caught
   end
-  -- no need to proceed if not left button or has no selection
-  if
-    ('left' ~= button)
-    or (not self.doc:has_selection())
-    or (1 < clicks)
+
+  -- no need to proceed if: not enabled, not left button, no selection
+  -- or if this is a multi-click event
+  if not config.plugins.dragdropselected.enabled
+    or 'left' ~= button
+    or not self.doc:has_selection()
+    or 1 < clicks
   then
       return on_mouse_pressed(self, button, x, y, clicks)
   end
+
   -- convert pixel coordinates to line and column coordinates
   local iLine, iCol = self:resolve_screen_position(x, y)
   -- get selection coordinates
-  local iSelLine1, iSelCol1, iSelLine2, iSelCol2 = self.doc:get_selection(true)
-  -- set flag for on_mouse_released and on_mouse_moved() methods to detect dragging
-  self.bClickedIntoSelection = isInSelection(iLine, iCol, iSelLine1, iSelCol1,
-                                             iSelLine2, iSelCol2)
-  if self.bClickedIntoSelection then
-    self.drag_start_loc = { x, y }
-    -- stash selection for inserting later
-    self.sDraggedText = self.doc:get_text(self.doc:get_selection())
-    self.dragged_selection = { iSelLine1, iSelCol1, iSelLine2, iSelCol2 }
-  else
-    self.bClickedIntoSelection = nil
-    self.dragged_selection = nil
+  local iLine1, iCol1, iLine2, iCol2 = self.doc:get_selection(true)
```

Comment:
Save also the swapped status
```suggestion
  local iLine1, iCol1, iLine2, iCol2, swapped = self.doc:get_selection(true)
```
and pass it to `doc:set_selection`, so that the selection "order" is preserved.

Reply:
no, it didn't help https://github.com/SwissalpS/liteXLplugins/tree/fixDragDropSelectedSwapped

I have started work on multi-selection scenarios though ;)

Reply:
What I meant is the location of the caret in relations to the selection:
![Schermata del 2023-06-19 16-37-50](https://github.com/lite-xl/lite-xl-plugins/assets/2798487/c5b21510-1f19-4e07-aef4-05c9c5bfa8ce)
vs
![Schermata del 2023-06-19 16-37-56](https://github.com/lite-xl/lite-xl-plugins/assets/2798487/080f5c70-14f3-4674-8814-50ee3638eef9)
both when starting the drag, and when applying it.

And your changes seem to fix this.

Reply:
not using ``Doc:get_selection()`` anymore. Wasn't accurate when user had multiple selections.